config/action.d/ipthreat.conf

```diff
@@ -0,0 +1,99 @@
+#
+# Reporting an IP is a serious action. Make sure that it is legit.
+# Consider using this action only for:
+#   * IP that has been banned more than once
+#   * High max retry to avoid user mis-typing password
+#   * Filters that are unlikely to be human error
+#
+# This action relies on `ipthreat_apikey` being added to the very bottom of this action conf,
+# and the appropriate flags and system set.
+#
+# Example, for ssh bruteforce (in section [sshd] of `jail.local`):
+# ```
+#   action = %(known/action)s
+#            ipthreat[ipthreat_flags="8",ipthreat_system="SSH"]
+# ```
+#
+# `ipthreat_system` is a short name of the system attacked, i.e. SSH, SMTP, MYSQL, PHP, etc.
+#
+# For `ipthreat_flags`, most cases will use 8 (BruteForce), but you could use others.
+# You can use the name or the ordinal.
+# Multiple values are comma separated.
+# ```
+# Name           Ordinal Description
+# Dns            1       Abuse/attack of dns (domain name server)
+# Fraud          2       General fraud, whether orders, misuse of payment info, etc
+# DDos           4       Distributed denial of service attack, whether through http requests, large ping attack, etc
+# BruteForce     8       Brute force login attack
+# Proxy          16      IP is a proxy like TOR or other proxy server
+# Spam           32      Email, comment or other type of spam
+# Vpn            64      IP is part of a VPN
+# Hacking        128     General hacking outside of brute force attack (includes vulnerability scans, sql injection, etc.). Use port scan flag instead if it's just probe on ports.
+# BadBot         256     Bad bot that is not honoring robots.txt or just flooding with too many requests, etc
+# Compromised    512     The ip has been taken over by malware or botnet
+# Phishing       1024    The ip is involved in phishing or spoofing
+# Iot            2048    The ip has targetted an iot (Internet of Things) device
+# PortScan       4096    Port scan
+# See https://ipthreat.net/bulkreportformat for more information
+# ```
+
+[Definition]
+
+# bypass action for restored tickets
+norestored = 1
+
+# Option:  actionstart
+# Notes.:  command executed on demand at the first ban (or at the start of Fail2Ban if actionstart_on_demand is set to false).
+# Values:  CMD
+#
+actionstart =
+
+# Option:  actionstop
+# Notes.:  command executed at the stop of jail (or at the end of Fail2Ban)
+# Values:  CMD
+#
+actionstop =
+
+# Option:  actioncheck
+# Notes.:  command executed once before each actionban command
+# Values:  CMD
+#
+actioncheck =
+
+# Option:  actionban
+# Notes.:  command executed when banning an IP. Take care that the
+#          command is executed with Fail2Ban user rights.
+#
+# Tags:    See jail.conf(5) man page
+# Values:  CMD
+#
+actionban = curl -sSf "https://api.ipthreat.net/api/report" -X POST -H "Content-Type: application/json" -H "X-API-KEY: <ipthreat_apikey>" -d '{"ip":"<ip>","flags":"<ipthreat_flags>","system":"<ipthreat_system>","notes":"fail2ban"}'
```

Comment:
This will be only substituted correctly if all of the interpolation variables are simple tokens.
If some of them become some special chars, fail2ban would prefer variable escape, so it would be not substituted inside the single quote-string and you can get `"system":"$f2bV_ipthreat_system"` instead of `"system":"..."`.
So possibly something like this would be better:
```diff
- -d '{"ip":"<ip>","flags":"<ipthreat_flags>","system":"<ipthreat_system>","notes":"fail2ban"}'
+ -d "{\"ip\":\"<ip>\",\"flags\":\"<ipthreat_flags>\",\"system\":\"<ipthreat_system>\",\"notes\":\"fail2ban\"}"
```

Also I wonder if ipthreat_system cannot simply contain `<name>` (name of jail) by default. E.g. set this in Init section:
```ini
[Init]
ipthreat_system = <name>
```

Reply:
I have performed these changes, please let me know if they are sufficient

Reply:
I think it's OK.